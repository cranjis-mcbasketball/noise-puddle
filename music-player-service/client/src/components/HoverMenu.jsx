import React from "react";
import HoverDiv from "../styled_components/hover_menu_style.js";
import Button_Clear from "../styled_components/buttons/clear_button.js";
import Hover_Next from "../styled_components/Hover_Next.js";

const HoverMenu = props => (
  <HoverDiv>
    <Hover_Next>
      Next Up <Button_Clear>Clear</Button_Clear>
    </Hover_Next>

    {props.playList.map((song, id) => {
      return (
        <div key={id}>
          <h4>Track Number</h4>
          <p>{song.id}</p>
          <h4>Title</h4>
          <p>{song.song_title}</p>
          <h4>Artist</h4>
          <p>{song.artist_name}</p>
          <h4>Album Cover</h4>
          <img src={song.album_cover} width="27" height="27" />
        </div>
      );
    })}
  </HoverDiv>
);

export default HoverMenu;
