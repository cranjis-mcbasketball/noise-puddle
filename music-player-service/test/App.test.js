import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount } from "enzyme";
import MusicPlayerOnFooter from "../client/src/components/MusicPlayer.jsx";
import ButtonPlay_Pause from "../client/src/components/MusicPlayer.jsx";
import HoverMenu from "../client/src/components/HoverMenu.jsx";

describe("Examining the syntax of Jest tests", () => {
  it("sums numbers", () => {
    expect(1 + 2).toEqual(3);
    expect(2 + 2).toEqual(4);
  });
});

describe("MusicPlayer", () => {
  test("should render snapshot", () => {
    const wrapper = mount(<MusicPlayer />);
    expect(wrapper).toMatchSnapshot();
  });

  test("song information should render.", () => {
    const songMock = [
      {
        id: "Test #1",
        song_title: "I'm baby",
        artist_name: "Queen",
        album_cover:
          "https://i.kym-cdn.com/photos/images/original/001/627/845/c22.jpg",
        music_url: "some Url#2"
      },
      {
        id: "Test #2",
        song_title: "uh.",
        artist_name: "Spears",
        album_cover:
          "https://i.kym-cdn.com/photos/images/original/001/627/845/c22.jpg",
        music_url: "some URL #2"
      }
    ];

    const example = shallow(<HoverMenu playList={songMock} />);
    expect(example).toMatchSnapshot();
  });

  //on hover should render the menu
  test("Music Player should render properly", () => {
    const div = document.createElement("div");
    ReactDOM.render(<MusicPlayerOnFooter />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  const mockFn = jest.fn();

  test("playbutton simulating click event", () => {
    const wrapper = shallow(<ButtonPlay_Pause handClick={mockFn} />);
    wrapper.simulate("click");
    expect(mockFn).toHaveBeenCalled();
  });
});
