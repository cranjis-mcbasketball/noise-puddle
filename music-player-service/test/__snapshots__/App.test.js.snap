// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicPlayer should render snapshot 1`] = `
<MusicPlayer>
  <styled.div>
    <div
      className="sc-fzoLsD difmhU"
    >
      <audio
        onCanPlay={[Function]}
        onEnded={[Function]}
        onTimeUpdate={[Function]}
        preload="auto"
        src=""
        volume={0.5}
      />
      <div
        id="previous-div"
      >
         
        <styled.button
          className="previous-button"
          onClick={[Function]}
        >
          <button
            className="sc-AxhUy iUflkL previous-button"
            onClick={[Function]}
          >
            <img
              height="30%"
              src="https://image.flaticon.com/icons/svg/254/254437.svg"
              width="30px"
            />
          </button>
        </styled.button>
      </div>
      <div
        id="play-pause-div"
      >
         
        <styled.button
          className="playPause-button"
          onClick={[Function]}
        >
          <button
            className="sc-AxhCb laQNKB playPause-button"
            onClick={[Function]}
          >
            <img
              height="30%"
              src="https://image.flaticon.com/icons/svg/254/254434.svg"
              width="30px"
            />
          </button>
        </styled.button>
      </div>
      <div
        id="next-div"
      >
         
        <styled.button
          className="next-button"
          onClick={[Function]}
        >
          <button
            className="sc-AxgMl gCkUFs next-button"
            onClick={[Function]}
          >
            <img
              height="30%"
              src="https://image.flaticon.com/icons/svg/254/254428.svg"
              width="30px"
            />
          </button>
        </styled.button>
      </div>
      <div
        id="shuffle-div"
      >
         
        <styled.button
          className="shuffle-button"
          onClick={[Function]}
        >
          <button
            className="sc-AxheI cjZMoI shuffle-button"
            onClick={[Function]}
          >
            <img
              height="30%"
              src="https://image.flaticon.com/icons/svg/2057/2057590.svg"
              width="30px"
            />
          </button>
        </styled.button>
      </div>
      <div
        id="repeat-div"
      >
        <styled.button
          className="repeat-button"
          onClick={[Function]}
        >
          <button
            className="sc-Axmtr giAfMv repeat-button"
            onClick={[Function]}
          >
            <img
              height="30%"
              src="https://image.flaticon.com/icons/svg/565/565272.svg"
              width="30px"
            />
          </button>
        </styled.button>
      </div>
      <styled.img
        height="27"
        id="queen-image"
        src=""
        width="27"
      >
        <img
          className="sc-fzokOt dhrFRT"
          height="27"
          id="queen-image"
          src=""
          width="27"
        />
      </styled.img>
      <styled.small
        id="start-time"
      >
        <small
          className="sc-fznKkj iVXrZs"
          id="start-time"
        />
      </styled.small>
      <span
        id="seek-container"
      >
        <styled.div>
          <div
            className="sc-fzpans eJsimJ"
          >
            <progress
              max="100"
              onClick={[Function]}
              value={0}
            />
          </div>
        </styled.div>
      </span>
      <styled.small
        id="end-time"
      >
        <small
          className="sc-fznZeY fRlhXC"
          id="end-time"
        />
      </styled.small>
      <div>
        <styled.button
          className="volume-button"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="sc-fzozJi hRaFMj volume-button"
            onClick={[Function]}
            onMouseOver={[Function]}
          >
            <img
              height="30%"
              src="https://image.flaticon.com/icons/svg/727/727269.svg"
              width="30px"
            />
          </button>
        </styled.button>
      </div>
      <div
        className="playlist-div"
      >
        <styled.button
          className="playList-button"
          onClick={[Function]}
        >
          <button
            className="sc-AxmLO hhKiZn playList-button"
            onClick={[Function]}
          >
            <img
              height="30%"
              src="https://image.flaticon.com/icons/svg/565/565266.svg"
              width="30px"
            />
          </button>
        </styled.button>
      </div>
    </div>
  </styled.div>
</MusicPlayer>
`;

exports[`MusicPlayer song information should render. 1`] = `
<styled.div>
  <styled.div>
    Next Up 
    <styled.button>
      Clear
    </styled.button>
  </styled.div>
  <div
    key="0"
  >
    <h4>
      Track Number
    </h4>
    <p>
      Test #1
    </p>
    <h4>
      Title
    </h4>
    <p>
      I'm baby
    </p>
    <h4>
      Artist
    </h4>
    <p>
      Queen
    </p>
    <h4>
      Album Cover
    </h4>
    <img
      height="27"
      src="https://i.kym-cdn.com/photos/images/original/001/627/845/c22.jpg"
      width="27"
    />
  </div>
  <div
    key="1"
  >
    <h4>
      Track Number
    </h4>
    <p>
      Test #2
    </p>
    <h4>
      Title
    </h4>
    <p>
      uh.
    </p>
    <h4>
      Artist
    </h4>
    <p>
      Spears
    </p>
    <h4>
      Album Cover
    </h4>
    <img
      height="27"
      src="https://i.kym-cdn.com/photos/images/original/001/627/845/c22.jpg"
      width="27"
    />
  </div>
</styled.div>
`;
